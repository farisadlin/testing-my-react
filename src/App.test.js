// myForm.test.js
import React from "react";
import { render, screen, waitFor, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'
import userEvent from "@testing-library/user-event";

import { MyForm } from "./components/myForm.js";

let renderLogin, handleSubmit;

beforeEach(() => {
  handleSubmit = jest.fn();
  renderLogin = render(<MyForm withSubmit={handleSubmit} />);
});

afterEach(cleanup);

describe("render logic correctly", () => {
  it("rendering and submitting a basic Formik form", async () => {
    userEvent.type(renderLogin.getByLabelText(/first name/i), "John");
    userEvent.type(renderLogin.getByLabelText(/last name/i), "Dee");
    userEvent.type(
      renderLogin.getByLabelText(/email/i),
      "john.dee@someemail.com"
    );

    userEvent.click(renderLogin.getByRole("button", { text: /submit dong/i }));

    await waitFor(() => {
      expect(handleSubmit).toHaveBeenCalledTimes(1);
      expect(handleSubmit).toHaveBeenCalledWith({
        email: "john.dee@someemail.com",
        firstName: "John",
        lastName: "Dee",
      });
    });
  });
});

describe("renders UI correctly", () => {
  it("render whole UI with snapshot", () => {
    const tree = renderer.create(<MyForm  />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})