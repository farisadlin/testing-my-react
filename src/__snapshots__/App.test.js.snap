// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI correctly render whole UI with snapshot 1`] = `
<div>
  <h1>
    Sign Up
  </h1>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <label
      htmlFor="firstName"
    >
      First Name
    </label>
    <input
      id="firstName"
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Jane"
      value=""
    />
    <label
      htmlFor="lastName"
    >
      Last Name
    </label>
    <input
      id="lastName"
      name="lastName"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Doe"
      value=""
    />
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="jane@acme.com"
      type="email"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
